src/encoder.rs

```diff
@@ -177,6 +177,11 @@ impl<W: Write> Encoder<W> {
     ///
     /// Note: This function also writes a control extension if necessary.
     pub fn write_frame(&mut self, frame: &Frame) -> Result<(), EncodingError> {
+        if frame.buffer.len() > 0 {
+            assert_ne!(frame.width, 0, "frame width is zero; perhaps you forgot to fill it in?");
+            assert_ne!(frame.height, 0, "frame height is zero; perhaps you forgot to fill it in?");
+        }
```

Comment:
Is the result an illegal `gif` because such frames are disallowed or is it illegal because we encode it incorrectly? In either case it seems improper to *panic*. The former should result in an error return, the latter should be fixed by a correct frame or an error return if we're just not capable at the moment.

Reply:
Hm, maybe the solution then would be to fix the frame? The encoder does know about the image's width and height after all. Would something like
```rust
        let width = if !frame.buffer.is_empty() && frame.width == 0 {
            self.width
        } else {
            frame.width
        };
        let height = if !frame.buffer.is_empty() && frame.height == 0 {
            self.height
        } else {
            frame.height
        };
        writer.write_le(width)?;
        writer.write_le(height)?;
```
make more sense?

Reply:
Guessing a fixup is almost never truly helpful. It extends the 'allowed' inputs and just causes callers to be more sloppy because it's suddenly part of the expected behavior. I would refuse the temptation to guess, and only fixup parameters if there's truly only *one* defined possibility and even then with scrutiny.

Reply:
Mmh, I see here what you mean with preferring to have explicit behavior here. I gather your preferred solution would be to return an error code, but what should the error condition be? The obvious answer would be `frame.buffer.len() == width * height`, but I couldn't help but notice this in the documentation for `frame.buffer` (emphasis mine):
> Buffer containing the image data. **Only indices unless configured differently.**

What does "configured differently" mean? (also the docs should be updated to document these conditions, but I can do that in a separate PR if you can point me to some GIF docs with this described)

Reply:
Given compatibility constraints, how does it sound if I would only check if there's at least enough data and then pa

Reply:
Yeah, but then that defeats the point - with the default size being zero, passing any array (except empty) will work.